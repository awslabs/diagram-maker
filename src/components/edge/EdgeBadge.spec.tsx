import * as Preact from 'preact';
import { shallow } from 'preact-render-spy';

import { EdgeBadge } from '.';

describe('EdgeBadge', () => {

  const destroyCallback = jest.fn();
  const renderCallback = jest.fn((container: HTMLElement): HTMLElement => {
    const element = document.createElement('div');
    element.textContent = 'Example edge content';
    container.appendChild(element);
    return element;
  });

  it('renders badge if passed', () => {
    const testId = 'testBadge';
    const edge = shallow(
      <EdgeBadge
        id={testId}
        renderCallback={renderCallback}
        destroyCallback={destroyCallback}
        src={{ x:0, y:100 }}
        dest={{ x:100, y:0 }}
      />
    );

    expect(edge).toMatchSnapshot();
  });

  it('renders badge when isPartOfEdgePair is true', () => {
    const testId = 'overlappingBadge';
    const edge = shallow(
      <EdgeBadge
        id={testId}
        renderCallback={renderCallback}
        destroyCallback={destroyCallback}
        src={{ x:0, y:100 }}
        dest={{ x:100, y:0 }}
        isPartOfEdgePair={true}
      />
    );

    expect(edge).toMatchSnapshot();
  });
});
