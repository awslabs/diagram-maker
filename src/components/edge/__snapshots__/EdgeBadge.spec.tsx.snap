// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EdgeBadge renders badge if passed 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-badge"
  style="left: 0px; top: 0px; width: 100px; height: 100px;"
>
  <div
    data-id="testBadge"
    data-type="DiagramMaker.EdgeBadge"
    data-event-target={true}
    class="dm-badge-inner"
  >
    <ComposeView
      renderCallback={[Function mockConstructor]}
      destroyCallback={[Function mockConstructor]}
    >
    </ComposeView>
  </div>
</div>

`;

exports[`EdgeBadge renders badge when isPartOfEdgePair is true 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-badge"
  style="left: -25px; top: -25px; width: 100px; height: 100px;"
>
  <div
    data-id="overlappingBadge"
    data-type="DiagramMaker.EdgeBadge"
    data-event-target={true}
    class="dm-badge-inner"
  >
    <ComposeView
      renderCallback={[Function mockConstructor]}
      destroyCallback={[Function mockConstructor]}
    >
    </ComposeView>
  </div>
</div>

`;
